samples/sample_hevc_fei/src/ref_list_manager.cpp

```diff
@@ -885,3 +886,169 @@ void EncodeOrderControl::ReleaseResources(HevcTask & task)
     std::for_each(task.m_preEncOutput.begin(), task.m_preEncOutput.end(), Release);
     task.m_preEncOutput.clear();
 }
+
+// EncodeOrderExtControl : public EncodeOrderControl
+HevcTask* EncodeOrderExtControl::ReorderFrame(mfxFrameSurface1 * surface)
+{
+    HevcTask* free_task = nullptr;
+
+    if (surface)
+    {
+        if (!m_free.empty())
+        {
+            free_task = &m_free.front();
+            m_reordering.splice(m_reordering.end(), m_free, m_free.begin());
+
+            msdk_atomic_inc16((volatile mfxU16*)&surface->Data.Locked);
+        }
+        else
+        {
+            assert(!"No free task in pool");
+        }
+    }
+
+    if (free_task)
+    {
+        free_task->Reset();
+
+        free_task->m_surf = surface;
+        std::vector<Frame>::const_iterator fi = FrameByDisplayOrder(m_frameOrder);
+        if (fi == m_FrameInfo.cend())
+            assert(!"No file entry for given frame order");
```

Comment:
In release build this code would do nothing. You have to add some other error processing. We use exception in rest of the code

Reply:
"assert(x) is a macro that evaluates to nothing when NDEBUG is defined". Is it?